### PR TITLE
Document recoverable error handling with `ResultError`

### PR DESCRIPTION
**Pull request by kuhar**
Add a subsection on recoverable error handlign to the Coding Standards doc.

### CODE REVIEW

**Comment by kuhar** [comment]
@dstutt @amdrexu @Flakebi @vettoreldaniele PTAL

**Comment by amdvlk-admin** [comment]
### Test summary for commit b1620202cb29d87d49ed096468331e8af94dc9cf
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 44abc48d199dd28369bf7415eee1751c4bf616d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev b8ed024d536d1bebb35927a0f4f470dca2f2dfc6</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev cdd74831d061bf65fdf5af4a0b6ce99fe2c558a9</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/218612)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32354/54653 (59.2%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22299/54653 (40.8%)</li><li>Warnings: 0/54653 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32194/54653 (58.9%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22458/54653 (41.1%)</li><li>Warnings: 1/54653 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32530/54653 (59.5%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22123/54653 (40.5%)</li><li>Warnings: 0/54653 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35511/54653 (65.0%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 19142/54653 (35.0%)</li><li>Warnings: 0/54653 (0.0%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — APPROVED** [review]

File: docs/CodingStandards.md [hunk 1]
```diff
@@ -196,6 +198,48 @@ be meaningfully ignored, e.g.:
...
+[recoverable error handling](https://llvm.org/docs/ProgrammersManual.html#recoverable-errors).
+In short, functions that either return a value or fail should return `llvm::Expected<T>`,
+while functions that do not return any value should have `llvm::Error` for the return type.
+
+LLPC has its own concrete error type, `Llpc::ResultError`, which bridges
+`llvm::Error`/`Expected<T>` and `Vkgc::Result`. In addition to being compatible with the
+usual LLVM error handling utilities like `llvm::cantFail`, `llvm::handleErrors`, etc.,
+we allow `llvm::Error`s to be converted to `Vkgc::Result` using
+`result = Llpc::reportError(std::move(err))` and `result = Llpc::errorToResult(std::move(err))`.
+See [`llpcError.h`](llpc/util/llpcError.h) for more details.
```
Comment (Flakebi) [inline docs/CodingStandards.md:215]: This link should point to `../llpc/util/llpcError.h`.

**Review by Flakebi — COMMENTED** [review]
Looks good to me, I left a nit inline.
Thanks for all your work on better error messages.

File: docs/CodingStandards.md (hunk 1 above)
Comment (kuhar) [reply docs/CodingStandards.md]: Thanks, fixed.

**Review by Flakebi — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4f75e142d665af138cad0df6323e20ea09a7de9d
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 44abc48d199dd28369bf7415eee1751c4bf616d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev b8ed024d536d1bebb35927a0f4f470dca2f2dfc6</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev cdd74831d061bf65fdf5af4a0b6ce99fe2c558a9</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/218612)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32354/54653 (59.2%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22299/54653 (40.8%)</li><li>Warnings: 0/54653 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32194/54653 (58.9%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22458/54653 (41.1%)</li><li>Warnings: 1/54653 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32530/54653 (59.5%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22123/54653 (40.5%)</li><li>Warnings: 0/54653 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35511/54653 (65.0%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 19142/54653 (35.0%)</li><li>Warnings: 0/54653 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by kuhar** [comment]
Ping for a merge @amdrexu @JaxLinAMD

**Review by amdrexu — APPROVED** [review]

**Pull request merged**